lib/r2d2/android_pay_token.rb

```diff
@@ -0,0 +1,27 @@
+module R2D2
+  class AndroidPayToken
+    include Util
+
+    attr_accessor :encrypted_message, :ephemeral_public_key, :tag
+
+    def initialize(token_attrs)
+      self.ephemeral_public_key = token_attrs["ephemeralPublicKey"]
+      self.tag = token_attrs["tag"]
+      self.encrypted_message = token_attrs["encryptedMessage"]
+    end
+
+    def decrypt(private_key_pem)
+      private_key = OpenSSL::PKey::EC.new(private_key_pem)
+
+      shared_secret = generate_shared_secret(private_key, ephemeral_public_key)
+
+      # derive the symmetric_encryption_key and mac_key
+      hkdf_keys = derive_hkdf_keys(ephemeral_public_key, shared_secret, 'Android')
+
+      # verify the tag is a valid value
+      verify_mac(hkdf_keys[:mac_key], encrypted_message, tag)
+
+      JSON.parse(decrypt_message(encrypted_message, hkdf_keys[:symmetric_encryption_key]))
```

Comment:
I kept the JSON parse here. In addition to my arguments outlined in https://github.com/spreedly/r2d2/pull/10#discussion_r188063881 - now with the factory method:

```ruby
R2D2.build_token(android_or_google_pay_attrs, recipient_id_etc).decrypt(private_key_pem) 
=> {"dpan"=>"4895370012003478", "expirationMonth"=>12} # etc, always a hash
```

Reply:
Agreed, using the built 'json' library is fine.